test/list_test.go

```diff
@@ -12,194 +12,167 @@ import (
 
 // TestList is the unit test for List operation.
 func TestList(t *testing.T) {
+	g := NewWithT(t)
+
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
 	client, _ := newKine(ctx, t)
 
-	t.Run("ListSuccess", func(t *testing.T) {
+	keys := shuffleList([]string{"/key/1", "/key/2", "/key/3", "/key/4", "/key/5"})
```

Comment:
nit: you could print the shuffled list before executing the tests. That way, you know the order of those keys in case of failure.